native-platform/src/main/java/net/rubygrapefruit/platform/file/FileSystemInfo.java

```diff
@@ -37,6 +37,20 @@
     @ThreadSafe
     String getFileSystemType();
 
+    /**
+     * Returns true if additional details are known for the file system.
+     * <p>
+     * You must not query additional details if they are unknown.
+     * <p>
+     * Additional details are:
+     * <ul>
+     *     <li>{@link #isCasePreserving()}</li>
+     *     <li>{@link #isCaseSensitive()}</li>
+     * </ul>
+     */
+    @ThreadSafe
+    boolean isDetailsKnown();
```

Comment:
As this is part of the public API, I think it's important to have nice names, and `isDetailsKnown()` has a singular/plural collision. Maybe `isCaseSensitivityKnown()` would be a way to work around it?